pennylane/labs/resource_estimation/templates/subroutines.py

```diff
@@ -56,3 +61,79 @@ def resource_params(self) -> dict:
     def resource_rep(cls, num_wires) -> CompressedResourceOp:
         params = {"num_wires": num_wires}
         return CompressedResourceOp(cls, params)
+
+
+class ResourceQuantumPhaseEstimation(qml.QuantumPhaseEstimation, ResourceOperator):
+    """Resource class for QPE"""
+
+    # TODO: Add a secondary resource decomp which falls back to op.pow_resource_decomp
+
+    @staticmethod
+    def _resource_decomp(
+        base_class, base_params, num_estimation_wires, **kwargs
+    ) -> Dict[CompressedResourceOp, int]:
+        gate_types = {}
+
+        hadamard = ResourceHadamard.resource_rep()
+        adj_qft = ResourceAdjoint.resource_rep(ResourceQFT, {"num_wires": num_estimation_wires})
+        ctrl_op = ResourceControlled.resource_rep(base_class, base_params, 1, 0, 0)
+
+        gate_types[hadamard] = num_estimation_wires
+        gate_types[adj_qft] = 1
+        gate_types[ctrl_op] = (2**num_estimation_wires) - 1
+
+        return gate_types
+
+    def resource_params(self) -> dict:
+        op = self.hyperparameters["unitary"]
+        num_estimation_wires = len(self.hyperparameters["estimation_wires"])
+
+        return {
+            "base_class": type(op),
+            "base_params": op.resource_params(),
+            "num_estimation_wires": num_estimation_wires,
+        }
+
+    @classmethod
+    def resource_rep(
+        cls, base_class, base_params, num_estimation_wires, **kwargs
```

Comment:
I don't think we have any `**kwargs` for `resource_rep` so it is probably not needed here.